chips/nrf51/src/ble_advertising_driver.rs

```diff
@@ -88,19 +88,13 @@ pub const BLE_HS_ADV_TYPE_MFG_DATA: usize = 0xff;
 
 
 
+#[derive(Default)]
 pub struct App {
     // used for adv data
     app_write: Option<AppSlice<Shared, u8>>,
 }
 
-impl Default for App {
-    fn default() -> App {
-        App { app_write: None }
-    }
-}
-
-pub struct BLE<'a, R: BleAdvertisementDriver + 'a, A: hil::time::Alarm + 'a> {
-    radio: &'a R,
```

Comment:
If you kept this field in, but just changed the type to `&'a nrf51::radio::Radio` you wouldn't need to use the `unsafe` blocks you added.